repository.go

```diff
@@ -66,22 +66,44 @@ func newRepository(options repositoryOptions, channels repositoryChannels) *repo
 	}
 
 	repo.options.storage.Init(options.backupPath, options.appName)
-	// In the future, remove the dependency of the repository and just pass in the storage
-	repo.deltaProcessor = newDeltaProcessor(repo.options.storage, repo, channels)
+	repo.deltaProcessor = newDeltaProcessor(repo, channels)
+
+	if loadedState, err := repo.options.storage.Load(); err == nil && loadedState != nil {
+		repo.updateState(loadedState)
+	} else {
+		repo.featureState.Store(&FeatureMemoryState{
+			Features: make(map[string]*api.Feature),
+			Segments: make(map[int][]api.Constraint),
+		})
+	}
 
+	// Delta processor needs to be collapsed into this module, it's far too jealous of this domain at the moment
 	if repo.isStreaming {
 		repo.streamingClient = newStreamingClient(
 			options,
 			channels,
-			repo.deltaProcessor,
-		)
+			repo.deltaProcessor)
 	}
 
 	go repo.sync()
 
 	return repo
 }
 
+func (r *repository) updateState(features *api.FeatureResponse) {
```

Comment:
One place to control access to mutating the state. This isn't perfect and should be its own thing. But that requires some redesign of the repository layer